### PR TITLE
BatchSpanProcessor with dedicated thread.

### LINKED ISSUE #2454: BatchProcessing for Spans should use own thread by default
Need to do a similar improvement as already done for logs and metrics.
https://github.com/open-telemetry/opentelemetry-rust/pull/2436

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #2454 
## Changes
- The existing _runtime-based_ BatchSpanProcessor has been moved to separate module - `span_processor_with_async_runtime` - which is enabled behind the feature-flag: `experimental_trace_batch_span_processor_with_async_runtime`. This is consistent with the batch-processor in logs, and periodic-reader in metrics. Feel free to ignore this during review. It's more of movement from one-file to another.
- The _thread-based_ BatchSpanProcessor, is added inside span_processor module, and re-exported as the default batch processor. chunk-based concurrent export features (which is present in runtime-based BatchSpanProcessor is not implemented as of now. Will be added later. 

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/trace/span_processor.rs [hunk 1]
```diff
@@ -166,6 +170,201 @@ impl SpanProcessor for SimpleSpanProcessor {
...
+        let (sender, receiver) = sync_channel(1);
+        self.message_sender
+            .try_send(BatchMessageDedicatedThread::Shutdown(sender))
+            .map_err(|_| TraceError::Other("Failed to send Shutdown message".into()))?;
+
+        let result = receiver
+            .recv_timeout(self.shutdown_timeout)
+            .map_err(|_| TraceError::ExportTimedOut(self.shutdown_timeout))?;
+        if let Some(handle) = self.handle.lock().unwrap().take() {
+            handle.join().expect("Failed to join thread");
```
[the remark this answers is not in the document]
Comment (lalitb) [reply opentelemetry-sdk/src/trace/span_processor.rs:318]: Nice to have these basic issues getting caught :)

**Comment by lalitb** [comment]
TODO : most stuff should work, apart from integration tests. Will be looking into it.

File: opentelemetry-sdk/src/trace/provider.rs [hunk 2]
```diff
@@ -296,12 +295,8 @@ impl Builder {
     }
 
     /// The [`SpanExporter`] setup using a default [`BatchSpanProcessor`] that this provider should use.
-    pub fn with_batch_exporter<T: SpanExporter + 'static, R: RuntimeChannel>(
-        self,
-        exporter: T,
-        runtime: R,
-    ) -> Self {
-        let batch = BatchSpanProcessor::builder(exporter, runtime).build();
+    pub fn with_batch_exporter<T: SpanExporter + 'static>(self, exporter: T) -> Self {
```
Comment (cijothomas) [inline opentelemetry-sdk/src/trace/provider.rs:298]: this needs to called out in changelog.md file.

File: opentelemetry-zipkin/src/exporter/mod.rs [hunk 3]
```diff
@@ -165,13 +164,12 @@ impl ZipkinPipelineBuilder {
     /// Install the Zipkin trace exporter pipeline with a batch span processor using the specified
     /// runtime.
     #[allow(deprecated)]
-    pub fn install_batch<R: RuntimeChannel>(
+    pub fn install_batch(
```
Comment (cijothomas) [inline opentelemetry-zipkin/src/exporter/mod.rs:167]: please add this to changelog as this is breaking.

**Review by cijothomas — APPROVED** [review]
LGTM!
Few nit comments, we can fix in follow ups.

**Pull request merged**